src/operations.jl

```diff
@@ -1,5 +1,34 @@
-struct BoolTuple end
-@inline BoolTuple(args::Bool...) = args
+# iterator
+struct ComponentIterator{C<:Colorant}
+    c::C
+end
+
+eltype(::Type{ComponentIterator{C}}) where {T, C <: Colorant{T}} = T
+length(::ComponentIterator{C}) where {N, C <: ColorantN{N}} = N
+
+function Base.iterate(itr::ComponentIterator{C}, state::Int=0) where {N, C <: ColorantN{N}}
+    state < 0 && return nothing
+    state >= N && return nothing
+    state += 1
+    return (itr[state], state)
+end
+
+@inline function Base.getindex(itr::ComponentIterator{C}, i::Integer) where {N, C <: ColorantN{N}}
+    N > 0 && i == 1 && return comp1(itr.c)
+    N > 1 && i == 2 && return comp2(itr.c)
+    N > 2 && i == 3 && return comp3(itr.c)
+    N > 3 && i == 4 && return comp4(itr.c)
+    N > 4 && i == 5 && return comp5(itr.c)
+    throw(BoundsError(itr, i))
+end
+Base.axes(::ComponentIterator{C}) where {N, C <: ColorantN{N}} = (Base.OneTo(N),)
+Base.ndims(::Type{ComponentIterator{C}}) where {C} = 1
+function Base.broadcastable(itr::ComponentIterator{C}) where {T, N, C <: Colorant{T, N}}
+    (itr...,)::NTuple{N, T}
```

Comment:
I think this should return the iterator itself (`itr`) instead of the tuple, but I don't know how to handle that well. :confused: